src/microbe_stage/Microbe.Interior.cs

```diff
@@ -389,38 +389,36 @@ public Microbe Divide()
 
         var currentPosition = GlobalTransform.origin;
 
-        // Separate the two cells.
-        var separation = new Vector3(Radius, 0, 0);
+        // Find the direction to the right from where the cell is facing
+        var direction = GlobalTransform.basis.Quat().Normalized().Xform(Vector3.Right);
 
-        if (Colony != null)
-        {
-            // When in a colony we approximate a much higher separation distance
-            var colonyRadius = separation.x;
+        // Start calculating separation distance
+        var organellePositions = organelles!.Organelles.Select(o => Hex.AxialToCartesian(o.Position)).ToList();
 
-            foreach (var colonyMember in Colony.ColonyMembers)
-            {
-                if (colonyMember == this)
-                    continue;
+        float distanceRight = MathUtils.GetMaximumDistanceInDirection(Vector3.Right, Vector3.Zero, organellePositions);
+        float distanceLeft = MathUtils.GetMaximumDistanceInDirection(Vector3.Left, Vector3.Zero, organellePositions);
 
-                var radius = colonyMember.Radius + Constants.COLONY_DIVIDE_EXTRA_DAUGHTER_OFFSET;
-
-                // TODO: switch this to something else if this is too slow for large colonies
-                var positionInColony = colonyMember.GlobalTransform.origin - currentPosition;
+        if (Colony != null)
+        {
+            var colonyMembers = Colony.ColonyMembers.Select(c => c.GlobalTransform.origin);
 
-                var outerRadius = Math.Max(Math.Abs(positionInColony.x) + radius,
-                    Math.Abs(positionInColony.z) + radius);
+            distanceRight += MathUtils.GetMaximumDistanceInDirection(direction, currentPosition, colonyMembers);
+        }
 
-                if (outerRadius > colonyRadius)
-                    colonyRadius = outerRadius;
-            }
+        float width = distanceLeft + distanceRight + Constants.DIVIDE_EXTRA_DAUGHTER_OFFSET;
 
-            separation = new Vector3(colonyRadius + Constants.COLONY_DIVIDE_EXTRA_DAUGHTER_OFFSET, 0, 0);
-        }
+        if (CellTypeProperties.IsBacteria)
+            width *= 0.5f;
 
         // Create the one daughter cell.
-        var copyEntity = SpawnHelpers.SpawnMicrobe(Species, currentPosition + separation,
+        var copyEntity = SpawnHelpers.SpawnMicrobe(Species, currentPosition + direction * width,
             GetParent(), SpawnHelpers.LoadMicrobeScene(), true, cloudSystem!, spawnSystem!, CurrentGame);
 
+        // Since the daughter spawns right next to the cell, it should face the same way to avoid colliding
+        var daughterBasis = new Basis(Transform.basis.Quat().Normalized()).Scaled(copyEntity.Transform.basis.Scale);
```

Comment:
```suggestion
        var daughterBasis = new Basis(Transform.basis.Quat())
        {
            Scale = copyEntity.Transform.basis.Scale,
        };
```

I think that would be the optimal way. I didn't originally suggest copying the full transform from the parent, *just in case* the scales are different (which probably should only happen due to a bug but I think it's good to handle that sensibly anyway).